DRODLib/Character.cpp

```diff
@@ -4289,6 +4309,61 @@ bool CCharacter::IsEntityTypeAt(
 	return false;
 }
 
+//*****************************************************************************
+//Returns: how many of the specified game entity type (flags) are in rect (x,y,w,h)
+int CCharacter::CountEntityType(
+	const CCharacterCommand& command,
+	const CDbRoom& room,
+	const CSwordsman& player) const
+{
+	UINT px, py, pw, ph, pflags;  //command parameters
+	getCommandParams(command, px, py, pw, ph, pflags);
+	int count = 0;
+
+	if (player.wIdentity == pflags)
+	{
+		if (player.wX >= px && player.wX <= px + pw &&
+			player.wY >= py && player.wY <= py + ph)
+			++count;
+	}
+
+	for (UINT y = py; y <= py + ph; ++y) {
+		for (UINT x = px; x <= px + pw; ++x) {
+			CMonster* pMonster = room.GetMonsterAtSquare(x, y);
+			if (!pMonster) {
+				continue;
+			}
+
+			if (pMonster->wType == pflags) {
+				++count;
+				continue;
+			}
+
+			switch (wType) {
```

Comment:
I'm unclear of the origin of this 'wType' var. Same for the instance below.

Reply:
Thanks for catching. Accidentally used the local version of `wType` instead of `pMonster->wType` the first time and `pflags` the second time.